ledger/ledger-api-client/src/it/scala/com/digitalasset/ledger/client/CommandClientIT.scala

```diff
@@ -144,17 +148,19 @@ final class CommandClientIT
       .runWith(Sink.seq)
       .map(_.last) // one element is guaranteed
 
-  private def submitCommand(req: SubmitRequest): Future[Completion] =
+  private def submitCommand(req: SubmitRequest): Future[CompletionResponse] =
     commandClient().flatMap(_.trackSingleCommand(req))
 
   private def assertCommandFailsWithCode(
       submitRequest: SubmitRequest,
       expectedErrorCode: Code,
       expectedMessageSubString: String,
   ): Future[Assertion] =
-    submitCommand(submitRequest).map { completion =>
-      completion.getStatus.code should be(expectedErrorCode.value)
-      completion.getStatus.message should include(expectedMessageSubString)
+    submitCommand(submitRequest).map { result =>
+      inside(result) { case Left(NotOkResponse(_, grpcStatus)) =>
+        grpcStatus.code should be(expectedErrorCode.value)
+        grpcStatus.message should include(expectedMessageSubString)
+      }
```

Comment:
Probably just my ignorance about `Inside` but this match doesn't seem exhaustive and, if so, what (failure, hopefully) message do we get in other cases?

Reply:
Yes. So in case of not matching we will get a failure.
I based the example on scalatest docs:

https://www.scalatest.org/user_guide/using_inside

Checking the source code we get a failure when the match is partial:

```
   if (pf.isDefinedAt(value)) {
     ......
    }
    else {
      Inside.level.set(Inside.level.get - 1)
      throw new TestFailedException((_: StackDepthException) => Some(Resources.insidePartialFunctionNotDefined(value.toString())), None, pos)
    }
```

Reply:
Alright, it stringifies the whole value, which should be fine, thanks for checking and sorry for the laziness 👍🏼

Reply:
👍 I don't like the fact that this is not capture in the scaladocs themselves, had to dive into the source code to figure it out

Reply:
Sadly, it often happens with Scala libs.